lua/ollama.lua

```diff
@@ -0,0 +1,76 @@
+local utils = require 'utils'
+
+local Module = {}
+
+Module.name = 'ollama'
+
+Module.default_model = 'llama3.2:3b'
+
+Module.models = {
+  Module.default_model,
+  'deepseek-coder:1.3b',
+  'tinydolphin:latest',
+  'qwen2.5:7b',
+}
+
+local system_prompt = [[
+Return only the command to be executed as a raw string, no string delimiters
+wrapping it, no yapping, no markdown, no fenced code blocks, what you return
+will be passed to vim directly.
+
+Example 1: if the user asks: select abc
+
+You return only: /abc
+
+Example 2: If the user asks how to quit vim, you return: :q
+Here is the question:
+]]
+
+
+Module.append_prompt = function(prompt)
+  if not prompt then
+    error("Prompt is nil; please provide a valid prompt.")
+  end
+  return system_prompt .. prompt
+end
+
+Module.insert_command = function(cmd)
+  -- Ensure cmd is a string and trim any whitespace
+  cmd = tostring(cmd):gsub("^%s*(.-)%s*$", "%1")
+
+  local escaped = vim.fn.escape(cmd, '\\\"')
+
+  local no_colon = string.gsub(escaped, ":", "")
+
+  vim.fn.input(":", no_colon)
+end
+
+
+Module.prompt = function(model, prompt)
+  if not prompt then
+    error("Prompt is nil; please provide a valid prompt.")
+  end
```

Comment:
I liked that in a previous version you had this at the beginning of `exit.lua#prompt` as:

```lua
  if not prompt or prompt == "" then
    error("Prompt is nil; please provide a valid prompt.")
  end
```

It makes sense as it would cover an empty prompt for all the adapters.